src/Application/Actions/Hooks/StripeUpdate.php

```diff
@@ -78,12 +77,20 @@ protected function action(): Response
         return $this->respondWithData($event->data->object);
     }
 
-    public function handlePaymentIntentSucceeded(Event $event, $donation): Response
+    public function handleChargeSucceeded(Event $event)
     {
+        /** @var Donation $donation */
+        $donation = $this->donationRepository->findOneBy(['transactionId' => $event->data->object->payment_intent]);
+
+        if (!$donation) {
+            throw new DomainRecordNotFoundException('Donation not found');
+        }
+        
         // For now we support the happy success path,
         // as this is the only event type we're handling right now,
         // convert status to the one SF uses.
         if ($event->data->object->status === 'succeeded') {
+            $donation->setChargeId($event->data->object->id);
```

Comment:
@NoelLH do you think it's worth sending the charge Id to SF and storing it there as well, or you think it'll be cleaner to just have payment intent Id to identify a donation on SF side?

Reply:
I don't think it really needs to be in SF too @tbg-jasonfung as long as MatchBot can link them up. We can always check the object associations in Stripe admin.